packages/blockchain-wallet-v4-frontend/src/middleware/analyticsMiddleware/types.ts

```diff
@@ -0,0 +1,221 @@
+enum AnalyticsKey {
+  AMOUNT_SWITCHED = 'Amount Switched',
+  BUY_AMOUNT_ENTERED = 'Buy Amount Entered',
+  BUY_AMOUNT_MAX_CLICKED = 'Buy Amount Max Clicked',
+  BUY_AMOUNT_MIN_CLICKED = 'Buy Amount Min Clicked',
+  // BUY_LEARN_MORE_CLICKED = 'Buy Learn More Clicked', // doesn't exist on the web
+  BUY_PAYMENT_METHOD_SELECTED = 'Buy Payment Method Selected',
+  BUY_SELL_CLICKED = 'Buy Sell Clicked',
+  BUY_SELL_VIEWED = 'Buy Sell Viewed',
+  CARD_REJECTED = 'Card Rejected',
+  DASHBOARD_CLICKED = 'Dashboard Clicked',
+  DASHBOARD_VIEWED = 'Dashboard Viewed',
+  EMAIL_VERIFICATION_REQUESTED = 'Email Verification Requested',
+  SIGNED_IN = 'Signed In',
+  SIGNED_OUT = 'Signed Out',
+  SWAP_ACCOUNTS_SELECTED = 'Swap Accounts Selected',
+  SWAP_AMOUNT_ENTERED = 'Swap Amount Entered',
+  SWAP_AMOUNT_MAX_CLICKED = 'Swap Amount Max Clicked',
+  SWAP_AMOUNT_MIN_CLICKED = 'Swap Amount Min Clicked',
+  SWAP_CLICKED = 'Swap Clicked',
+  SWAP_FROM_SELECTED = 'Swap From Selected',
+  SWAP_RECEIVE_SELECTED = 'Swap Receive Selected',
+  SWAP_REQUESTED = 'Swap Requested',
+  SWAP_VIEWED = 'Swap Viewed',
+  UPGRADE_VERIFICATION_CLICKED = 'Upgrade Verification Clicked'
+}
+
+enum AnalyticsType {
+  EVENT = 'EVENT',
+  VIEW = 'VIEW'
+}
+
+type BasePayload = {
+  id: string
+  nabuId: string
+  originalTimestamp: string
+  type: AnalyticsType
+}
+
+type PageViewPayload = {
+  path: string
+  referrer: string
+  search: string
+  title: string
+  url: string
+}
+
+type AmountSwitchedPayload = BasePayload & {
+  product: 'SAVINGS' | 'SIMPLEBUY' | 'SWAP'
+  switch_to: 'CRYPTO' | 'FIAT'
+}
+
+type BuyAmountEnteredPayload = BasePayload & {
+  input_amount: number
+  input_currency: string
+  max_card_limit: number
+  output_currency: string
+  platform: 'WALLET'
+}
+
+type BuyAmountMaxClickedPayload = BasePayload & {
+  input_currency: string
+  max_card_limit: number
+  output_currency: string
+  platform: 'WALLET'
+}
+
+type BuyAmountMinClickedPayload = BasePayload & {
+  input_currency: string
+  output_currency: string
+  platform: 'WALLET'
+}
+
+type BuyPaymentMethodSelectedPayload = BasePayload & {
+  payment_type: 'BANK_ACCOUNT' | 'BANK_TRANSFER' | 'FUNDS' | 'PAYMENT_CARD'
+  platform: 'WALLET'
+}
+
+type BuySellClickedPayload = BasePayload & {
+  origin: 'BUY_WIDGET' | string
+  // type: "BUY" | "SELL"
+}
+
+type BuySellViewedPayload = BasePayload &
+  PageViewPayload & {
+    // type: "BUY" | "SELL"
```

Comment:
empty object ok to have here? Should probably just remove it.

Reply:
This is for v2, I've added it here just to be already here when I have to implement in 2 weeks.